DependencyInjection/JMSDiExtraExtension.php

```diff
@@ -53,6 +53,27 @@ public function load(array $configs, ContainerBuilder $container)
         $container->setParameter('jms_di_extra.cache_dir', $config['cache_dir']);
         $container->setParameter('jms_di_extra.disable_grep', $config['disable_grep']);
         $container->setParameter('jms_di_extra.doctrine_integration', $config['doctrine_integration']);
+
+        $config['service_naming_strategy'] = array_merge(
+            array(
+                'id' => 'jms_di_extra.service_naming_strategy.default',
+                'class' => null,
+            ),
+            isset($config['service_naming_strategy']) ? $config['service_naming_strategy'] : array()
+        );
+
+        if ($config['service_naming_strategy']['class'] !== null) {
+            $container->register(
+                'jms_di_extra.service_naming_strategy',
+                $config['service_naming_strategy']['class']
+            );
+        } else {
+            $container->setAlias(
+                'jms_di_extra.service_naming_strategy',
+                new Alias($config['service_naming_strategy']['id'])
+            );
+        }
+
```

Comment:
This whole block can be much more simplified because of the `service_naming_strategy` change below ;)